bluemira/equilibria/fem_fixed_boundary/fem_magnetostatic_2D.py

```diff
@@ -183,3 +200,255 @@ def calculate_b(self):
         self.B = dolfin.project(dolfin.as_vector((Bx, Bz)), w)
 
         return self.B
+
+
+class FemGradShafranovFixedBoundary(FemMagnetostatic2d):
+    """
+    A 2D fem Grad Shafranov solver. The solver is thought as support for the fem fixed
+    boundary module.
+
+    Parameters
+    ----------
+    mesh : Union[dolfin.mesh, str]
+        Filename of the xml file with the mesh definition or a dolfin mesh
+    boundaries : Union[dolfin.mesh, str]
+        Filename of the xml file with the boundaries definition or a MeshFunction that
+        defines the boundaries
+    p_order : int
+        Order of the approximating polynomial basis functions
+    """  # noqa (W505)
+
+    def __init__(self, mesh, boundaries=None, p_order=3):
+        super().__init__(mesh, boundaries, p_order)
+        self.k = 1
+
+    @property
+    def psi_ax(self):
+        """Poloidal flux on the magnetic axis"""
+        return np.max(self.psi.vector()[:])
+
+    @property
+    def psi_b(self):
+        """Poloidal flux on the boundary"""
+        return 0.0  # np.min(self.psi.vector()[:])
```

Comment:
Your call! I did this for sanity while playing around with stuff and forgot to remove... just realising that if somebody passes a different BC into the solve (see above comment) which isn't 0.0, this could cause a massive problem! Sorry for messing this up :face_with_head_bandage:

Reply:
A fix is in #1404